host/libcontainer_backend.go

```diff
@@ -602,85 +636,71 @@ func (l *LibvirtLXCBackend) Run(job *host.Job, runConfig *RunConfig, rateLimitBu
-				},
+	ifaceName, err := netutils.GenerateIfaceName("veth", 4)
+	if err != nil {
+		return err
+	}
+	if job.Config.HostNetwork {
+		// allow host network jobs to configure the network
+		config.Capabilities = append(config.Capabilities, "CAP_NET_ADMIN")
+	} else {
+		config.Hostname = hostname
+		config.Namespaces = append(config.Namespaces, configs.Namespace{Type: configs.NEWNET})
+		config.Networks = []*configs.Network{
+			{
+				Type:    "loopback",
+				Address: "127.0.0.1/0",
+				Gateway: "localhost",
 			},
-			Consoles: []lt.Console{{Type: "pty"}},
-		},
-		Resource: &lt.Resource{
-			Partition: "/machine/" + job.Partition,
-		},
-		OnPoweroff: "preserve",
-		OnCrash:    "preserve",
+			{
+				Type:              "veth",
+				Name:              "eth0",
+				Bridge:            l.bridgeName,
+				Address:           initConfig.IP,
+				Gateway:           initConfig.Gateway,
+				Mtu:               1500,
+				HostInterfaceName: ifaceName,
+			},
+		}
 	}
 	if spec, ok := job.Resources[resource.TypeMemory]; ok && spec.Limit != nil {
-		domain.Memory = lt.UnitInt{Value: *spec.Limit, Unit: "bytes"}
+		config.Cgroups.Resources.Memory = *spec.Limit
 	}
 	if spec, ok := job.Resources[resource.TypeCPU]; ok && spec.Limit != nil {
-		domain.CPUTune = &lt.CPUTune{Shares: milliCPUToShares(*spec.Limit)}
+		config.Cgroups.Resources.CpuShares = milliCPUToShares(*spec.Limit)
 	}
 
-	if !job.Config.HostNetwork {
-		domain.Devices.Interfaces = []lt.Interface{{
-			Type:   "network",
-			Source: lt.InterfaceSrc{Network: l.bridgeName},
-		}}
-	}
-
-	// attempt to run libvirt commands multiple times in case the libvirt daemon is
-	// temporarily unavailable (e.g. it has restarted, which sometimes happens in CI)
-	log.Info("defining domain")
-	var vd libvirt.VirDomain
-	if err := l.withConnRetries(func() (err error) {
-		vd, err = l.libvirt.DomainDefineXML(string(domain.XML()))
-		return
-	}); err != nil {
-		log.Error("error defining domain", "err", err)
-		return err
-	}
-	defer vd.Free()
-
-	log.Info("creating domain")
-	if err := l.withConnRetries(vd.Create); err != nil {
-		log.Error("error creating domain", "err", err)
-		return err
-	}
-	log.Info("getting domain uuid")
-	uuid, err := vd.GetUUIDString()
+	c, err := l.factory.Create(job.ID, config)
 	if err != nil {
-		log.Error("error getting domain uuid", "err", err)
 		return err
 	}
-	l.state.SetContainerID(job.ID, uuid)
 
-	domainXML, err := vd.GetXMLDesc(0)
-	if err != nil {
-		log.Error("error getting domain xml", "err", err)
-		return err
+	process := &libcontainer.Process{
+		Args: []string{"/.containerinit", job.ID},
+		User: "root",
+	}
+	if job.Config.TTY {
+		console, err := process.NewConsole(0, 0)
+		if err != nil {
+			return err
+		}
+		// copy the console to /dev/null because if we don't
+		// do anything with the console, it will be destroyed
+		// by the garbage collector and the job will get a
+		// tty hangup.
+		go io.Copy(ioutil.Discard, console)
```

Comment:
This is weird, can we disassociate the console from the flynn-host process somehow?